components/camera/camera_waitgroup.go

```diff
@@ -1,49 +1,48 @@
-package videosource
+package camera
 
 import (
 	"context"
 	"sync"
 
 	"github.com/edaniels/gostream"
 
-	"go.viam.com/rdk/components/camera"
 	"go.viam.com/rdk/pointcloud"
 	"go.viam.com/rdk/rimage/transform"
 )
 
 // CameraWaitGroup is a wrapper for camera.Camera with a sync.WaitGroup.
 type CameraWaitGroup struct {
-	cam                     camera.Camera
-	activeBackgroundWorkers sync.WaitGroup
+	Cam                     Camera
+	ActiveBackgroundWorkers sync.WaitGroup
```

Comment:
Why is `ActiveBackgroundWorkers` exported now?

Reply:
I moved CameraWaitGroup to camera package, so ActiveBacgroundWorers needs to be public for access in videosource.

See webcam.go diff.

Reply:
can we find a way to not have to do that? This feels like a bit of a footgun now based on how someone else may incorrectly use this encapsulated concept

Reply:
Just circling back on this PR. 

Semantically, I think it makes sense for `WaitGroupCamera` to be in the `camera` package instead of `videosource`.

It also could be useful for future use cases - if more child routines are needed off the camera component.

@bazile-clyde What do you think? Any ideas on how to keep WaitGroup private?

I added a warning comment above the definition.

Reply:
You could move the retry logic into the `CamerWaitGroup` struct since it handles the `sync.WaitGroup` stuff. The `DoCommand` handles arbitrary data. So maybe the objects the retry logic needs could go in there. Something like

```
func (c *CameraWaitGroup) DoCommand(ctx context.Context, cmd map[string]interface{}) (map[string]interface{}, error) {
	if _, ok := cmd["reconnection"]; ok {
		return c.doReconnection(ctx, cmd)	
	}
	return c.cam.DoCommand(ctx, cmd)
}

```
could work. You'd then create a map of the args you need in `NewWebcamSource`, create a `WaitGroup` then call the `DoCommand` on it before returning. The reconnection logic would then be completely encapsulated in this class. 

Furthermore, you could make `WaitGroup` fulfill the `ProxyType` interface to make this struct easier to unwrap with `UnwrapProxy`.

@edaniels Let me know if this solution abuses what the `DoCommand` was created for. Seems pretty generic.

Reply:
Just saw this. I think I ended up doing something similar but also removed some reconfigurable logic that was causing issues in https://github.com/viamrobotics/rdk/pull/1883

Reply:
Cool. @SeanAvery you could wait until that PR is merged, rebase, and then fix this problem. It should be a lot simpler.